### PR TITLE
Add group name to asset

### PR DESCRIPTION
**Pull request by shalabhc**
### Summary & Motivation

Exploring the addition of a groupName attribute on assets.

### How I Tested These Changes

Added 1 test case.

Also locally in graphql

image

### CODE REVIEW

File: python_modules/dagster/dagster/core/asset_defs/assets.py [hunk 1]
```diff
@@ -62,6 +63,7 @@ def __init__(
             f"expected keys: {all_asset_keys}",
         )
         self._resource_defs = check.opt_mapping_param(resource_defs, "resource_defs")
+        self._group_names = check.opt_mapping_param(group_names, "group_name")
```
Comment (sryza) [inline python_modules/dagster/dagster/core/asset_defs/assets.py:66]: nitpick: group_name -> group_names

File: python_modules/dagster/dagster/core/asset_defs/decorators.py [hunk 2]
```diff
@@ -125,6 +127,7 @@ def asset(
             Frameworks may expect and require certain metadata to be attached to a op. Values that
             are not strings will be json encoded and must meet the criteria that
             `json.loads(json.dumps(value)) == value`.
+        group_jname (Optional[str]): A string name used to organize multiple assets into groups.
```
Comment (sryza) [inline python_modules/dagster/dagster/core/asset_defs/decorators.py:130]: group_jname -> group_name

File: python_modules/dagster/dagster/core/asset_defs/decorators.py [hunk 3]
```diff
@@ -282,6 +289,7 @@ def multi_asset(
     partition_mappings: Optional[Mapping[str, PartitionMapping]] = None,
     op_tags: Optional[Dict[str, Any]] = None,
     can_subset: bool = False,
+    # TODO - does this need a group name too?
```
Comment (clairelin135) [inline python_modules/dagster/dagster/core/asset_defs/decorators.py:292]: I think we don't need to add a `group_name` field to `multi_asset` for now. People use `@multi_asset` much more rarely than `@asset`, and this field can be easily added later

**Review by clairelin135 — COMMENTED** [review]
Thanks Shalabh, this is looking great. Adding a couple small comments inline.

You'll need to run `make generate_graphql` in the dagster root directory to fix BK tests and to update the graphql schema

**Review by clairelin135 — APPROVED** [review]
LGTM!

**Pull request merged**